File: examples/assertion-scoring-override/override_scoring.js
// Override scoring function that focuses on grammar and fluency
module.exports = (namedScores, context) => {
  console.log('Override scoring function (JavaScript):', namedScores);

  // Prioritize grammar and fluency over accuracy
  const accuracyScore = namedScores.accuracy || 0;
  const fluencyScore = (namedScores.fluency || 0) * 1.5;
  const grammarScore = (namedScores.grammar || 0) * 1.5;

  const totalScore = (accuracyScore + fluencyScore + grammarScore) / 3;

  return {
    pass: totalScore >= 0.7,
    score: totalScore,
    reason: `Style-focused score: accuracy: ${accuracyScore}, fluency(1.5x): ${fluencyScore}, grammar(1.5x): ${grammarScore}`,
  };
};

File: examples/assertion-scoring-override/override_scoring.py
def func(named_scores, context=None):
    print('Override scoring function (Python):', named_scores)
    
    # Focus only on accuracy, ignore style metrics
    accuracy_score = named_scores.get('accuracy', 0) * 2
    
    # Simplified scoring that only cares about accuracy
    total_score = min(accuracy_score, 1.0)
    
    return {
        'pass': total_score >= 0.7,
        'score': total_score,
        'reason': f'Accuracy-focused score (2x weight): {accuracy_score}',
    } 